### PR TITLE
add test case for not setting schema history to empty

### PR DESCRIPTION
**Pull request by Haijuncao**
dup of #1703 as the merge is kind of messed up, so start a new branch.

### CODE REVIEW

File: src/dbnode/storage/database_test.go [hunk 1]
```diff
@@ -704,7 +704,6 @@ func TestDatabaseUpdateNamespaceSchemaNotSet(t *testing.T) {
 	mapCh <- nsMap
 
 	// wait till the update has propagated
-	<-updateCh
 	time.Sleep(10 * time.Millisecond)
```
Comment (richardartoul) [inline src/dbnode/storage/database_test.go]: does this sleep do anything anymore?

Comment (Haijuncao) [reply src/dbnode/storage/database_test.go]: <-updateCh is flaky, sometimes it block forever
I found that wait for 10ms is good enough for etcd change to show up.
If we fix the flaky <-updateCh, then we can remove the sleep.
 
Either block wait, or time wait, we don't need both.
